raft.go

```diff
@@ -1229,7 +1229,7 @@ func stepLeader(r *raft, m pb.Message) error {
 
 				if refused != "" {
 					r.logger.Infof("%x ignoring conf change %v at config %s: %s", r.id, cc, r.prs.Config, refused)
-					m.Entries[i] = pb.Entry{Type: pb.EntryNormal}
```

Comment:
Switching to normal entry is trying to increase applied index so that the next confChange can be applicable. It seems that it is right move to return dropped

Reply:
> Switching to normal entry is trying to increase applied index so that the next confChange can be applicable.

I don't think switching to normal entry will make a difference here. But I agree returning proposal dropped early is helpful.

Reply:
You are right. The raft will update applied index even if the apply fails. I was thinking that the normal is used to catch up with pendingConfIndex so that it won't impact the following confChange.

Reply:
aha. Thanks for explanation! I think next `ProposeConfChange()` will succeed once the `raft.Applied` is advanced by etcd raftNode `r.Advance()`

Reply:
For reference, the use of normal entry was added [here](https://github.com/etcd-io/raft/commit/cb095791f970bd0a2017785680de3fd440524059) 
I think the idea was to support mixed list of entries (ConfChanges and Normal) in proposal, but is it even possible to get mixed list right now? 
When I check for creation of `pb.Message{Type: pb.MsgProp`, there are two distinct places:
1. regular [Propose](https://github.com/etcd-io/raft/blob/65a0bf3b6779e6b84d6c032bde1fcdf7a49d13df/node.go#L444)
2. [confChangeToMsg](https://github.com/etcd-io/raft/blob/65a0bf3b6779e6b84d6c032bde1fcdf7a49d13df/node.go#L456)  - used in [ProposeConfChange](https://github.com/etcd-io/raft/blob/65a0bf3b6779e6b84d6c032bde1fcdf7a49d13df/node.go#L464) and [raft.appliedTo](https://github.com/etcd-io/raft/blob/5fe1c31c515840dd807b8f07d91f704f8334a1b6/raft.go#L696)